src/distributions/uniform.jl

```diff
@@ -0,0 +1,35 @@
+"""
+    Uniform(collection)
+
+Create a categorical distribution over a collection of objects.
+"""
+mutable struct Uniform{C, T}
+    collection::C
+    _set::Union{Set{T}, Nothing} # keep track of what's in the collection to make pdf more efficient
+end
+
+Uniform(c) = Uniform{typeof(c), eltype(c)}(c, nothing)
+Uniform(c::Set) = Uniform{typeof(c), eltype(c)}(c, c)
+
+rand(rng::AbstractRNG, d::Uniform) = rand(rng, d.collection)
```

Comment:
Unclear about the rationale here. If you are allowing for non-sets and converting it into sets and giving them equal weights, then how can you sample from the collection which might have multiple instances of the same and therefore higher weight?

Reply:
Yeah, thanks, I need to redesign this.